integreat_cms/summ_ai_api/summ_ai_api_client.py

```diff
@@ -80,20 +98,19 @@ async def translate_text_field(self, session, text_field):
                 },
             ) as response:
                 # Wait for the response
-                response_data = await response.json()
-                # Check whether the text was translated successfully
-                if "translated_text" not in response_data:
-                    if "error" in response_data:
-                        raise SummAiException(
-                            f"API error: {response.status} - {response_data['error']}"
-                        )
-                    raise SummAiException(
-                        f"Unexpected API result: {response.status} - {response_data!r}"
-                    )
-                # Let the field handle the translated text
-                text_field.translate(response_data["translated_text"])
-                return text_field
-        except (aiohttp.ClientError, asyncio.TimeoutError, SummAiException) as e:
+                try:
+                    response_data = await response.json()
+                except aiohttp.ContentTypeError as e:
+                    raise SummAiInvalidJSONError(
+                        f"API delivered invalid JSON: {response.status} - {await response.text()}"
+                    ) from e
```

Comment:
I suggest adding this logging because the message of this exception itself is not logged.
If execution falls into this "except" block, there is not clue in the log at all to what the problem was.

Use case example:
If API returns 404 - in this case, the response body is usually html --> json can't be parsed --> but you'd never guess it from the log without explicit logging

```suggestion
                    logger.error(
                        "Failed to parse JSON from API response: %s",
                        response.status,
                    )
                    raise SummAiInvalidJSONError from e
```

Reply:
Do you have a specific intention in leaving out the `await response.text()` from the logging?

Reply:
Yes, but I don't insist here.

I assumed that the status code will give us enough information about the problem, while the response text may not be as useful in cases where it is not JSON. 
_(I decided so by looking at the repeated html 'not found' in the log)_

Reply:
I decided to keep the exception text and the `await response.text()` (but only there), without separating it out into one call to `….text()` and parsing it ourselves, because it would basically be duplicated code from a foreign library that would only be run in the event of an error, which shouldn't be often and thus bring no big performance improvement.

Furthermore, while the resulting string of `….text()` is not being cached, if I interpret the client library correctly, the only substantial wait time in calling it arises from not having received the whole response yet. This means that calling it a second time should not result in any significant delay, since the byte reader can read the whole data sequence in one go.